### ISSUE TITLE
ganon-build, Errorcode: -4

### ISSUE THREAD

**Issue by mat43672** [pirovc/ganon#56]
Hello,

I am trying to include Ganon in the benchmarking platform we currently developing (https//lemmi.ezlab.org). In short, it means wrapping it in a docker to both build a reference using specific genomes, and then run an analysis reusing this database.

We managed to make it work for Ganon on different servers and laptops, however it crashes on the server we use for the actual benchmarking runs... seems related to the kernel or install of this machine, as docker is not a complete virtualization. 

The error I get is 

```
Running taxonomic clustering (TaxSBP.py)...
169 bins created. Done. Elapsed time: 5.922635078430176 seconds.
Approximate (upper-bound) # unique k-mers: 5999982
Bloom filter calculated size with fp<=0.05: 896.5887MB (39172558 bits/bin * 192 optimal bins [169 real bins])
Building index (ganon-build)...
The following command failed to execute:
/ganon/build/ganon-build -e /bbx/tmp/DB_tmp/acc_bin.txt --filter-size-bits 7521131136 -k 19 -n 3 -t 32 -o /bbx/tmp/DB.filter --verbose   /bbx/tmp/All_seqs.fna
Errorcode: -4
Error:
--seqid-bin-file      /bbx/tmp/DB_tmp/acc_bin.txt
--output-filter-file  /bbx/tmp/DB.filter
--filter-size         896
--filter-size-bits    7521131136
--hash-functions      3
--kmer-size           19
--n-batches           1000
--n-refs              400
--threads             32
--update-filter-file
--update-complete     0
--verbose             1
--reference-files
                      /bbx/tmp/All_seqs.fna


Traceback (most recent call last):
  File "/ganon/ganon", line 596, in <module>
    main()
  File "/ganon/ganon", line 195, in main
    stdout, stderr, errcode = run(run_ganon_build_cmd, print_stderr=True)
  File "/ganon/ganon", line 472, in run
    if errcode!=0: raise Exception()
Exception
```

Probably not sufficient to give me an answer, unless a parameter is wrong, but I would be glad to provide extra detail if I knew what to look for.

Of note, when building the docker container on a different machine, ganon-build -h leads to a core dump. When building the container on the incriminated server (following: Installing GCC7 in a separate environment with conda), ganon-build -h does display the help. An improvement, but the main error keeps happening unfortunately.

Any suggestion for a debug strategy is welcome.

Thanks

Mathieu

**Comment by pirovc** [comment]
Hi @mat43672 

It seens like it's all properly configured as you said. I'd guess it's a compilation issue. First thing I can think of is to try to remove `-msse4.2` from https://github.com/pirovc/ganon/blob/3a0b87fcda1f47455b4eefc6c47989b0c601f16f/CMakeLists.txt#L77

Then re-run cmake and make. Please let me know the outcome.

Best 
Vitor

**Comment by mat43672** [comment]
It solved the problem, thanks for the quick and efficient solution !

**Comment by mdehollander** [comment]
:+1: Looking forward to see some benchmarks in Lemmi

**Comment by pirovc** [comment]
@mat43672 good to know. 

Just FYI, there are optimizations on SeqAN lib for that flag. Either your architecture supports a different version of it or it does not support it at all, which can lead to a decrease in performance but I guess it won't be critical.

**Comment by mat43672** [comment]
Ok I keep that in mind and will ensure we are using this lib as best as possible. I know that even when using docker, architecture is affecting the perfs of each method.

**Comment by pirovc** [comment]
Solved on #75
